app/models/concerns/host_common.rb

```diff
@@ -67,9 +67,12 @@ def url_for_boot(file)
 
   def puppetca?
     return false if self.respond_to?(:managed?) and !managed?
-    !!(puppet_ca_proxy and puppet_ca_proxy.url.present?)
+    puppetca_exists?
   end
 
+  def puppetca_exists?
+    !!(puppet_ca_proxy and puppet_ca_proxy.url.present?)
+  end
```

Comment:
Nitpick, please add a blank line in after this method.  (No spaces/tabs, just blank.)